{
  "contract": "ether_lotto",
  "tool": "mythril",
  "start": 1565777014.7536612,
  "end": 1565777020.4775863,
  "duration": 5.7239251136779785,
  "analysis": {
    "error": null,
    "issues": [
      {
        "address": 288,
        "code": "assert(msg.value == TICKET_AMOUNT)",
        "debug": "The exception is triggered under the following conditions:\n\ncallvalue: 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff5\ncalldata_EtherLotto_0: 0x93e84cd900000000000000000000000000000000000000000000000000000000\ncalldatasize_EtherLotto: 0x4\n",
        "description": "A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that `assert()` should only be used to check invariants. Use `require()` for regular input checking. ",
        "filename": "/dataset/time_manipulation/ether_lotto.sol",
        "function": "play()",
        "lineno": 36,
        "title": "Exception state",
        "type": "Informational"
      },
      {
        "address": 298,
        "code": "pot += msg.value",
        "debug": "storage_1: 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc\ncalldata_EtherLotto_0: 0x93e84cd900000000000000000000000000000000000000000000000000000000\ncalldatasize_EtherLotto: 0x4\ncallvalue: 0xa\n",
        "description": "A possible integer overflow exists in the function `play()`.\nThe addition or multiplication may result in a value higher than the maximum representable integer.",
        "filename": "/dataset/time_manipulation/ether_lotto.sol",
        "function": "play()",
        "lineno": 39,
        "title": "Integer Overflow ",
        "type": "Warning"
      },
      {
        "address": 436,
        "code": "bank.transfer(FEE_AMOUNT)",
        "debug": "",
        "description": "In the function `play()` the following predictable state variables are used to determine Ether recipient:\n- block.timestamp\n",
        "filename": "/dataset/time_manipulation/ether_lotto.sol",
        "function": "play()",
        "lineno": 49,
        "title": "Dependence on predictable environment variable",
        "type": "Warning"
      },
      {
        "address": 436,
        "code": "bank.transfer(FEE_AMOUNT)",
        "debug": "",
        "description": "Multiple sends exist in one transaction, try to isolate each external call into its own transaction. As external calls can fail accidentally or deliberately.\nConsecutive calls: \nCall at address: 511\n",
        "filename": "/dataset/time_manipulation/ether_lotto.sol",
        "function": "play()",
        "lineno": 49,
        "title": "Multiple Calls",
        "type": "Information"
      },
      {
        "address": 511,
        "code": "msg.sender.transfer(pot - FEE_AMOUNT)",
        "debug": "",
        "description": "A possible transaction order independence vulnerability exists in function play(). The value or direction of the call statement is determined from a tainted storage location",
        "filename": "/dataset/time_manipulation/ether_lotto.sol",
        "function": "play()",
        "lineno": 52,
        "title": "Transaction order dependence",
        "type": "Warning"
      }
    ],
    "success": true
  }
}